pkg/sql/plan2/build_function.go

```diff
@@ -100,14 +98,47 @@ func getFunctionExprByNameAndPlanExprs(name string, exprs []*Expr) (resultExpr *
 	return
 }
 
+func rewriteStarToCol(query *Query, node *Node) (string, error) {
+	if node.NodeType == plan.Node_TABLE_SCAN {
+		return node.TableDef.Cols[0].Name, nil
+	} else {
+		for _, child := range node.Children {
+			for _, col := range query.Nodes[child].ProjectList {
+				return col.ColName, nil
+			}
+		}
+	}
+	return "", errors.New(errno.InvalidColumnReference, "can not find any column when rewrite count(*) to starcount(col)")
+}
+
 func getFunctionExprByNameAndAstExprs(name string, astExprs []tree.Expr, ctx CompilerContext, query *Query, node *Node, binderCtx *BinderContext, needAgg bool) (resultExpr *Expr, isAgg bool, err error) {
-	name = strings.ToLower(name)
+	// name = strings.ToLower(name)
 	args := make([]*Expr, len(astExprs))
-	// deal with special function
+	// deal with special function [rewrite some ast function expr]
 	switch name {
 	case "extract":
+		// rewrite args[0]
 		kindExpr := astExprs[0].(*tree.UnresolvedName)
 		astExprs[0] = tree.NewNumVal(constant.MakeString(kindExpr.Parts[0]), kindExpr.Parts[0], false)
+	case "count":
+		// count(*) : astExprs[0].(type) is *tree.NumVal
+		// count(col_name) : astExprs[0].(type) is *tree.UnresolvedName
+		switch astExprs[0].(type) {
+		case *tree.NumVal:
+			// rewrite count(*) to starcount(col_name)
+			name = "starcount"
+			var countColName string
+			countColName, err = rewriteStarToCol(query, node)
```

Comment:
This rewrite is actually wrong.

create table t (a int);

Select count(*) from T and select count(a) from T are not the same.    The first one count null as well.

Reply:
yes，   count(*) is not equal count(col)

so @nnsgmsone add a new function   "starcount"  to deal with this case.

and these code will rewrite count(*) to starcount(col)